engine/lib/phx/src/render/tex2d_save.rs

```diff
@@ -2,17 +2,17 @@ use image::{DynamicImage, ImageBuffer};
 use internal::ConvertIntoString;
 
 #[no_mangle]
-pub extern "C" fn Tex2D_Save_Png(
+pub unsafe extern "C" fn Tex2D_Save_Png(
     path: *const libc::c_char,
     sx: i32,
     sy: i32,
     components: i32,
     data: *mut u8,
 ) -> bool {
-    tex2d_save_png(&path.as_str(), sx, sy, components, data)
+    tex2d_save_png(path.as_str(), sx, sy, components, data)
 }
 
-pub fn tex2d_save_png(path: &str, sx: i32, sy: i32, components: i32, data: *mut u8) -> bool {
+pub unsafe fn tex2d_save_png(path: &str, sx: i32, sy: i32, components: i32, data: *mut u8) -> bool {
```

Comment:
How come we're adding unsafe back to these things? I was hoping to keep unsafe blocks constrained where possible when the function is generally safe for the most part.